### DISCUSSION TITLE
Github pages

### DISCUSSION THREAD

**Discussion by RuiGene** [community/community#27794] [Codespaces] [answered]
Im trying to host a website on github rn. Ive made my repo and named it with github.io but I cant convert it to a github page. It only presents me with the option of upgrading to the pro package but i thought u could do it for free

**Comment by airtower-luna** [comment]
You do not have any *public* repository named anything with `.github.io`. With a free account you can publish pages only from public repositories.

**Reply by ghost** [reply]
I believe that is because the person has not gone to the setting part of their page and then scrolled down and on the left select pages. Then click choose a theme....then wait a bit and a page should pop up.

**Reply by ghost** [reply]
https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile

This might help as well

**Comment by iamfoysal** [comment]
https://username.github.io/reponame/
you can follow this doc :  [GitHub page](
https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site)

**Comment by EliasAfara** [comment, marked as answer]
> Im trying to host a website on github rn. Ive made my repo and named it with github.io but I cant convert it to a github page. It only presents me with the option of upgrading to the pro package but i thought u could do it for free

I think your issue might be because you repository visibility is private. Could you please double check?

**Reply by RuiGene** [reply]
tysm!

**Comment by jfullstackdev** [comment]
the user is progressing by now but not yet replying here 

![image](https://user-images.githubusercontent.com/47092464/182078475-0aade7af-8630-4add-ace5-e19cf3fdc60d.png)

**Comment by radar-blip** [comment]
Accomplished. Welcome to GitHub Pages! 👍🏼

**Comment by RuiGene** [comment]
Thanks for the help guys, was a dumb mistake on my part.

**Comment by ghost** [comment]
Hello
I built the site according to the guide. Create a repository with a username and upload the site template.
  The GitHub page has been activated in the repository!!
But after 1 hour, my site is still not published!
Could it be a problem with my template?
My template can be seen on my GitHub page.

**Reply by airtower-luna** [reply]
Take a look at the Actions tab in your repository, you can see the failing build there. It looks like the directory to deploy isn't configured correctly in your Pages settings.

If you need additional help please create a new discussion for your problem instead of getting into the (solved) discussion of someone else's problem. :slightly_smiling_face:

**Reply by radar-blip** [reply]
You need to add an index.html to your repo. Or at least a Readme.md
